src/javatests/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImplTest.java

```diff
@@ -123,27 +144,147 @@ public void getAllScriptNames_fail() throws Exception, SQLException {
+
+  @Test
+  public void executeScript_writeChunked_success() throws Exception {
+    scriptManager = new ScriptManagerImpl(scriptRunner, scriptsMap, sortingColumnsMap);
+    Connection connection = DriverManager.getConnection("jdbc:hsqldb:mem:db_3");
+    Statement baseStmt = connection.createStatement();
+    baseStmt.execute(
+        "CREATE Table TestTable ("
+            + "ID INTEGER,"
+            + "TIMESTAMPS TIMESTAMP(6) WITH TIME ZONE"
+            + ")");
+    // Insert in different orders to test that the later query does apply ORDER BY.
+    for (int i = 16; i > 10; i--) {
+      baseStmt.execute(
+          String.format(
+              "INSERT INTO TestTable VALUES (%d, TIMESTAMP '2008-08-08 20:08:%02d.007000'"
+                  + " AT TIME ZONE INTERVAL '0:00' HOUR TO MINUTE)",
+              i, i + 8));
+    }
+    for (int i = 0; i < 11; i++) {
+      baseStmt.execute(
+          String.format(
+              "INSERT INTO TestTable VALUES (%d, TIMESTAMP '2008-08-08 20:08:%02d.007000'"
+                  + " AT TIME ZONE INTERVAL '0:00' HOUR TO MINUTE)",
+              i, i + 8));
+    }
+    baseStmt.close();
+    connection.commit();
+    ImmutableList<String> expectedFiles =
+        ImmutableList.<String>builder()
+            .add("default_chunked-20080808T200808S007000-20080808T200810S007000_0.avro")
+            .add("default_chunked-20080808T200811S007000-20080808T200813S007000_1.avro")
+            .add("default_chunked-20080808T200814S007000-20080808T200816S007000_2.avro")
+            .add("default_chunked-20080808T200817S007000-20080808T200819S007000_3.avro")
+            .add("default_chunked-20080808T200820S007000-20080808T200822S007000_4.avro")
+            .add("default_chunked-20080808T200823S007000-20080808T200824S007000_5.avro")
+            .build();
+    DataEntityManager dataEntityManagerTmp = new DataEntityManagerTempTestImpl("tmpTest");
+
+    scriptManager.executeScript(
+        connection,
+        /*dryRun=*/ false,
+        sqlTemplateRenderer,
+        "default_chunked",
+        dataEntityManagerTmp,
+        3);
+
+    assertThat(
+            Files.walk(dataEntityManagerTmp.getAbsolutePath(""))
+                .filter(Files::isRegularFile)
+                .sorted()
+                .map(path -> path.getFileName().toString())
+                .collect(Collectors.toList()))
+        .isEqualTo(expectedFiles);
+    // Validate result details for the first and the last chunks.
+    DataFileReader<Record> reader;
+    reader =
+        getReaderFromAvroFile(
+            dataEntityManagerTmp.getAbsolutePath(
+                "default_chunked-20080808T200808S007000-20080808T200810S007000_0.avro"));
+    assertRecordEqualsExpected(reader.next(), 0, "2008-08-08T20:08:08.007000000Z");
+    assertRecordEqualsExpected(reader.next(), 1, "2008-08-08T20:08:09.007000000Z");
+    assertRecordEqualsExpected(reader.next(), 2, "2008-08-08T20:08:10.007000000Z");
+    assertFalse(reader.hasNext());
+    reader =
```

Comment:
Mixing "act" and "assert" steps together is hard to follow.
Actually, to me it seems like two tests – one to check the first chunk and second to check the last

Also, splitting this test makes it very obvious, that "arrange" step should be moved to helper (part with creating and populating the table)

Reply:
Per off-line discussion and post-discussion investigation, the conclusions are:

1. The "act" and "assert" steps are not mixed in this case; the naming of one helper function (getReaderFromAvroFile) however may cause such confusion, so I renamed this method.
2. The test is indeed too long with assertions on various aspects, thus I split it into two tests: one checking that all the files are present, the other checking that the contents in the first and last chunks are correct (with the assumption that the contents in the middle are unlikely to be incorrect in this case).